tests/collection/test_matches.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) QuantCo 2025-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+import polars as pl
+
+import dataframely as dy
+
+
+def test_collection_matches_itself() -> None:
+    class MySchema(dy.Schema):
+        foo = dy.Integer()
+
+    # First collection has one member
+    class MyCollection1(dy.Collection):
+        x: dy.LazyFrame[MySchema]
+
+    assert MyCollection1.matches(MyCollection1)
+
+
+def test_collection_matches_different_members() -> None:
+    class MySchema(dy.Schema):
+        foo = dy.Integer()
+
+    # First collection has one member
+    class MyCollection1(dy.Collection):
+        x: dy.LazyFrame[MySchema]
+
+    # Second has an additional member
+    class MyCollection2(MyCollection1):
+        y: dy.LazyFrame[MySchema]
+
+    # Should not match
+    assert not MyCollection1.matches(MyCollection2)
+
+
+def test_collection_matches_different_schemas() -> None:
+    # Two schemas that do not match
+    class MyIntSchema(dy.Schema):
+        foo = dy.Integer()
+
+    class MyStringSchema(dy.Schema):
+        foo = dy.String()
+
+    assert not MyIntSchema.matches(MyStringSchema), (
+        "Test schemas must not match for test setup to make sense"
+    )
```

Comment:
I don't think you need this, this functionality should really be covered by the schema tests.

Reply:
I did not mean this as a test that `Schema.matches` works (which indeed, should happen in the schema tests), I meant it as a sanity check for the setup of this specific test. The logic of the test is meant to be "if the schemas are different, collection-level comparison should fail", so I want to verify that I indeed created a setup that fulfills the first half of that sentence.